syz-fuzzer/machine_info.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+)
+
+func CollectMachineInfo() ([]byte, error) {
+	if runtime.GOOS != "linux" {
+		return []byte{}, nil
+	}
+
+	type machineInfoFunc struct {
+		name string
+		fn   func() ([]byte, error)
+	}
+
+	allMachineInfo := []machineInfoFunc{
+		{"CPU Info", readCPUInfo},
+		{"KVM", readKVMInfo},
+	}
+
+	var buffer bytes.Buffer
```

Comment:
bytes.Buffer is always used as a pointer, all methods are defined on the pointer type. If we will pass it around or call other function (fmt.Fprintf), we will need to take address and pass pointer.
It's better to do `buffer := new(bytes.Buffer)` to avoid taking address everywhere and using a mix of pointer and non-pointer type.